src/Service.Tests/SqlTests/RestApiTests/Patch/PatchApiTestBase.cs

```diff
@@ -585,16 +585,17 @@ public virtual async Task PatchWithNoPrimaryKeyRouteTest()
                     exceptionExpected: true,
                     expectedErrorMessage: RequestValidator.PRIMARY_KEY_NOT_PROVIDED_ERR_MESSAGE,
                     expectedStatusCode: HttpStatusCode.BadRequest
-                    );
+            );
         }
 
         /// <summary>
-        /// Test to validate that PATCH operation on inaccessible row (by virtue of database policy) fails.
+        /// Test to validate that PATCH operation fails because the database policy("@item.id ne 1234")
+        /// restricts modifying records where id is not 1234.
         /// </summary>
-        /// <returns></returns>
         [TestMethod]
         public virtual async Task PatchOneUpdateInAccessibleRowWithDatabasePolicy()
```

Comment:
If DB policy is present on Create operation, but not on an update. And the Patch results in an insert that is NOT allowed, what is our behavior in such a situation? do we have a test for this scenario?

Reply:
We don't have support for database policy for insert portion on upsert/upsertincremental operations yet. I opened an issue to have a discussion on the same : https://github.com/Azure/data-api-builder/issues/1275.